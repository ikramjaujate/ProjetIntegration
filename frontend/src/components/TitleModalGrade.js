import '../css/Grades.css';


const TitleModalGrade = ({id, bgColor, text}) => {
    return (
        <div className="modal-header row justify-content-center">
            <h5 id={id} className="modal-title p-1 shadow-sm rounded col-10 col-sm-11 col-lg-9" style={{backgroundColor:bgColor}}>{text}</h5>
        </div>
    );
}

export default TitleModalGrade;