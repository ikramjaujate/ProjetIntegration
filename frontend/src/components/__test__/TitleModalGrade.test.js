import React from 'react' ;
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16' ;
import 'jest-enzyme' ;
import 'jest-styled-components';
import TitleModalGrade from '../TitleModalGrade.js' ;

Enzyme.configure({ adapter: new Adapter() }); //Configurer Enzyme pour react 16


describe('TitleModalGrade', () => {
    it('Vérifier le texte affiché dans le titre', () => {
      const wrapper = shallow(<TitleModalGrade bgColor="" text="Directeur"></TitleModalGrade>);
      const paragraph = wrapper.find('h5');
      expect(paragraph).toHaveLength(1);
      expect(paragraph.text()).toEqual('Directeur');
    })
      
    it('Vérifier la couleur du fond du titre', () => {
      const wrapper = shallow(<TitleModalGrade bgColor="#086589" text=""/>);
      const paragraph = wrapper.find('h5.modal-title');
      expect(paragraph.prop('style')).toHaveProperty("backgroundColor", "#086589");
    })

    it("Vérifier que le props bgColor a une valeur par défaut", () => {
      const props = TitleModalGrade.defaultProps.bgColor;
      expect(props).toBeDefined();
      expect(props).toBe('var(--text-loading)');
    })

    it("Vérifier que le props text a une valeur par défaut", () => {
      const props = TitleModalGrade.defaultProps.text;
      expect(props).toBeDefined();
      expect(props).toBe('Chargement');
    })
})
  

  