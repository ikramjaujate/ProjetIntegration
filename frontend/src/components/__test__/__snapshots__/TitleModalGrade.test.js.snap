// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TitleModalGrade Snapshot - renders correctement 1`] = `
<div
  className="modal-header row justify-content-center"
>
  <h5
    className="modal-title test p-1 shadow-sm rounded col-10 col-sm-11 col-lg-9"
    style={
      Object {
        "backgroundColor": "#000000",
      }
    }
  >
    Test
  </h5>
</div>
`;
